daemon/health.go

```diff
@@ -136,9 +136,16 @@ func (p *cmdProbe) run(ctx context.Context, d *Daemon, cntr *container.Container
 		// Wait for probe to exit (it might take some time to call containerd to kill
 		// the process and we don't want dying probes to pile up).
 		<-execErr
+
+		var msg string
+		if out := output.String(); len(out) > 0 {
+			msg = fmt.Sprintf("Health check exceeded timeout (%v): %s", probeTimeout, out)
+		} else {
+			msg = fmt.Sprintf("Health check exceeded timeout (%v)", probeTimeout)
+		}
```

Comment:
So there was one concern I had with this patch (also why I move it separate from the other fix); we capture the full output of the exec'd process here, and basically that could be "anything".

I wonder if there's a need to limit the amount of output we capture (for situations where a health check does something really silly and produces XX MB of output)

Reply:
We do not capture the full output of the exec'd process: `output` is a `limitedBuffer` which only retains the first 4096 bytes. (IMO it would be more useful to retain the beginning _and end_ of output, discarding the middle, like `(os/exec.ExitError).Stderr`, but I digress...) The (truncated) output is already captured and returned for successful checks so this change is not breaking any new ground.

Reply:
>  output is a limitedBuffer 

Ah! 🤦‍♂️ you are so right. I didn't check. And it's there.. in plain sight, just a few lines up 🤦 https://github.com/moby/moby/blob/c9c55df1f25a10ff509ad4d56fd87f0172dee7a8/daemon/health.go#L95

> IMO it would be more useful to retain the beginning and end of output,

Yes! That's a bit of a pet-peeve of mine as well when things are truncated (stripping off start or end _quite often_ removes the relevant information). We could consider improving in future (that said, for this case, it's hard to consider "what parts" are important, as output could be "anything")).